app/views/declaration/confirmation/submission_confirmation_page.scala.html

```diff
@@ -34,16 +33,19 @@
   exitSurvey: exit_survey
 )
 
-@()(implicit request: Request[_], flash: Flash, messages: Messages)
+@()(implicit request: Request[_], messages: Messages)
 
-@lrnMessage(lrn: String) = {<div class="font-large">@messages("declaration.confirmation.lrn")</div> <div id="lrn-number" class="font-large bold">@lrn</div>}
+@lrnMessage(lrn: String) = {
+    <div class="font-large">@messages("declaration.confirmation.lrn")</div>
+    <div id="lrn-number" class="font-large bold">@lrn</div>
+}
 
 @decLink = @{
-    flash.get(FlashKeys.decId).map { decId =>
+    request.session.data.get(submission_uuid).map { declarationId =>
```

Comment:
Not something for now but we should decide on a name for this and apply it consistently `submissionId` or `declarationId` (confusing for it to be both imo)

Reply:
I had doubt about this. However it will be refactored in the upcoming ticket, 3464